src/engine.cpp

```diff
@@ -1246,6 +1246,10 @@ void CheckEngines()
 	for (const Engine *e : Engine::Iterate()) {
 		if (!e->IsEnabled()) continue;
 
+		if (e->type == VEH_TRAIN && e->u.rail.railveh_type == RAILVEH_WAGON) {
+			continue;
+		}
```

Comment:
Could use a comment, also the check can be one line:

```suggestion
		/* Don't consider train wagons, we need a powered engine available. */
		if (e->type == VEH_TRAIN && e->u.rail.railveh_type == RAILVEH_WAGON) continue;
```